packages/firestore/src/remote/backoff.ts

```diff
@@ -74,10 +83,13 @@ export class ExponentialBackoff {
 
   /**
    * Returns a promise that resolves after currentDelayMs, and increases the
-   * delay for any subsequent attempts.
+   * delay for any subsequent attempts. If there was a pending backoff operation
+   * already, it will be canceled.
    */
-  backoffAndWait(): Promise<void> {
-    const def = new Deferred<void>();
+  backoffAndRun(op: () => Promise<void>): void {
```

Comment:
`backoffAndRun` lets you schedule any type of operation here, but the ID already got assigned in the constructor. Should we move the assignment of the ID to here as well?